test/extensions/filters/listener/tls_inspector/tls_inspector_test.cc

```diff
@@ -79,6 +79,28 @@ class TlsInspectorTest : public testing::TestWithParam<std::tuple<uint16_t, uint
 #endif
   }
 
+  // Unlike mockSysCallForPeek, this method does not assume that the buffer is
+  // already sufficiently sized to hold the client hello. Instead it uses the
+  // buffers length to determine how much to copy.
+  void mockSysCallForPeekFillBuffer(std::vector<uint8_t>& client_hello) {
+#ifdef WIN32
+    EXPECT_CALL(os_sys_calls_, readv(_, _, _))
+        .WillOnce(Invoke(
+            [&client_hello](os_fd_t fd, const iovec* iov, int iovcnt) -> Api::SysCallSizeResult {
+              memcpy(iov->iov_base, client_hello.data(), iov->iov_len);
```

Comment:
Is it worth to assert the "iov->iov_len <= client_hello.size()"

Reply:
Thanks for pointing out, this made me realize we could just fold these two methods into one by just copying the min(client_hello_size, buffer length).